docs/source/basics/parameter_types.md

```diff
@@ -107,22 +145,48 @@ Finally, we currently support a single collection type which is `List[T]`, also
 provided by the `typing` module.
 
 ```python
-from typing import List, Optional
+from latch.types import LatchFile
+from typing import List
 
-@task
-def foo(
-  a: List[str] = ["AUG", "GGG"],
-  b: List[Optional[int]] = [1, None, 3, 4, None, 6]
+@workflow
+def rnaseq(
+    sample_identifiers: list[str],
+    sample_reads: List[LatchFile]
 ):
-  ...
+...
 ```
 
+![List](../assets/ui/list.png)
+
+### Dataclass
+
+If you want to handle file references and their associated metadata as inputs to your workflow, you may want to use a `dataclass`.
+
+```python
+from dataclasses import dataclass
+from typing import List
+
+@dataclass
+class Sample:
+    name: str
+    fastq: LatchFile
+
+@workflow
+def rnaseq(
+    samples: List[Sample]
+):
+
+```
+
+![List of class](../assets/ui/dataclass.png)
+
+### Custom Types
+
+If existing types don't directly work, it is possible to define your own custom types to be used in tasks and workflows. See the Flyte docs on how to do so [here](https://docs.flyte.org/projects/cookbook/en/latest/auto/core/extend_flyte/custom_types.html).
+
 ## Setting Default Values
 
-Notice that every parameter in the above list of types had a default value. This
-default value will be displayed to the user in the frontend interface and will
-be passed by default if no change is made. You do not have to provide a default
-value, but it helps document parameters and provide semantic context to users.
+If a parameter has a default value, it will be displayed to the user in the frontend interface and will be passed by default if no change is made. You do not have to provide a default value, but it helps document parameters and provide semantic context to users.
```

Comment:
- small clarifications
- don't want to encourage people setting defaults for *everything* since it will lead to stupid stuff like input files defaulting to "/input_file.fasta" or something like that which will be wrong
- expanded on reasons for giving default values
```suggestion
If a parameter has a default value, it is displayed to the user in the frontend interface and passed to the workflow if the user does not change it. Default values are optional but encouraged as, if a reasonable default exists, it can improve the user experience by encouraging best practices and allowing users to focus on the settings that really matter to them.
```